### PR TITLE
textgen-silence-output-feature in terminal

### PR DESCRIPTION
**Pull request by Abonia1**
Hello,
Added the new feature to silence TextGen's output in the terminal.

  - Description: Added a new feature to control printing of TextGen's output to the terminal., 
  - Issue: the issue #TextGen parameter to silence the print in terminal #10337  it fixes (if applicable)
  
  Thanks;

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
we should not be printing anything out period. not here at least. thats what we have callbacks for. i can update this to just delete

File: libs/langchain/langchain/llms/textgen.py [hunk 1]
```diff
@@ -184,13 +184,15 @@ def _call(
         prompt: str,
         stop: Optional[List[str]] = None,
         run_manager: Optional[CallbackManagerForLLMRun] = None,
+        print_output: bool = True,  # New parameter for printing control
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:187]:
```suggestion
```

File: libs/langchain/langchain/llms/textgen.py [hunk 2]
```diff
@@ -208,7 +210,8 @@ def _call(
                 prompt=prompt, stop=stop, run_manager=run_manager, **kwargs
             ):
                 combined_text_output += chunk.text
-            print(prompt + combined_text_output)
+            if print_output:  # Conditionally print the output
+                print(prompt + combined_text_output)
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:214]:
```suggestion
```

File: libs/langchain/langchain/llms/textgen.py [hunk 3]
```diff
@@ -220,7 +223,8 @@ def _call(
 
             if response.status_code == 200:
                 result = response.json()["results"][0]["text"]
-                print(prompt + result)
+                if print_output:  # Conditionally print the output
+                    print(prompt + result)
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:227]:
```suggestion
```

File: libs/langchain/langchain/llms/textgen.py [hunk 4]
```diff
@@ -353,6 +357,7 @@ async def _astream(
         prompt: str,
         stop: Optional[List[str]] = None,
         run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        print_output: bool = True,  # New parameter for printing control
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:360]:
```suggestion
```

File: libs/langchain/langchain/llms/textgen.py [hunk 5]
```diff
@@ -363,6 +368,7 @@ async def _astream(
         Args:
             prompt: The prompts to pass into the model.
             stop: Optional list of stop words to use when generating.
+            print_output (bool, optional): Whether to print the output to the terminal (default is True).
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:371]:
```suggestion
```

File: libs/langchain/langchain/llms/textgen.py [hunk 6]
```diff
@@ -413,6 +419,8 @@ async def _astream(
                     text=result["text"],
                     generation_info=None,
                 )
+                if print_output:  # Conditionally print the output
+                    print(prompt + chunk.text)
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:423]:
```suggestion
```

File: libs/langchain/langchain/llms/textgen.py [hunk 7]
```diff
@@ -191,6 +191,8 @@ def _call(
         Args:
             prompt: The prompt to use for generation.
             stop: A list of strings to stop generation when encountered.
+            print_output (bool, optional): Whether to print the output to the terminal.
+                Default is True.
```
Comment (hwchase17) [inline libs/langchain/langchain/llms/textgen.py:195]:
```suggestion
```

**Comment by Abonia1** [comment]
@hwchase17 Thanks for letting me know.

**Comment by baskaryan** [comment]
thanks @Abonia1!

**Pull request merged**